receiver/awscontainerinsightreceiver/internal/host/hostinfo.go

```diff
@@ -76,9 +81,21 @@ func NewInfo(awsSessionSettings awsutil.AWSSessionSettings, containerOrchestrato
 		containerOrchestrator: containerOrchestrator,
 		awsSessionCreator:     awsutil.GetAWSConfigSession,
 		nodeCapacityCreator:   newNodeCapacity,
-		ec2MetadataCreator:    newEC2Metadata,
-		ebsVolumeCreator:      newEBSVolume,
-		ec2TagsCreator:        newEC2Tags,
+		ec2MetadataCreator: withConfigurer(configurer, func(c *awsmiddleware.Configurer) func(context.Context, *session.Session, time.Duration, chan bool, chan bool, bool, int, *zap.Logger, ...ec2MetadataOption) ec2MetadataProvider {
```

Comment:
I think we need to break that line down and explain whats going on using helper functions, because currently you it is difficult to understand whats going on there without coming to this PR and finding this comment

Reply:
same thing with the return functions down below.